firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -480,18 +473,50 @@ public void testCannotReadAfterWriting() {
     assertNotNull(e);
   }
 
+  @Test
+  public void testReadandUpdateNonExistentDocumentWithExternalWrite() {
+    final FirebaseFirestore firestore = testFirestore();
+    AtomicInteger counter = new AtomicInteger(0);
+
+    // Make a transaction that will fail
+    Task<Void> transactionTask =
+        firestore.runTransaction(
+            transaction -> {
+              int count = counter.incrementAndGet();
+              // Get and update a document that doesn't exist so that the transaction fails.
+              DocumentReference doc = firestore.collection("nonexistent" + count).document();
```

Comment:
This code: 

```java
DocumentReference doc = firestore.collection("nonexistent").document();
```

(i.e. without the `+ count`) should give you a new random document every time. You shouldn't need to use different collections for each iteration.